chitchat/tests/cluster_test.rs

```diff
@@ -122,6 +180,49 @@ impl Simulator {
                         .add_link(node_1.gossip_advertise_addr, node_2.gossip_advertise_addr)
                         .await;
                 }
+                Operation::NodeStatusAssert {
+                    server_chitchat_id,
+                    chitchat_id,
+                    expected_status,
+                    timeout_opt,
+                } => {
+                    debug!(server_node_id=%server_chitchat_id.node_id, node_id=%chitchat_id.node_id, "node-presence-assert");
+                    let chitchat = self
+                        .node_handles
+                        .get(&server_chitchat_id)
+                        .unwrap()
+                        .chitchat();
+                    if let Some(timeout) = timeout_opt {
+                        let chitchat_id_clone = chitchat_id.clone();
+                        tokio::time::timeout(timeout, async move {
+                            let mut wait_printed = false;
+                            loop {
+                                let chitchat_guard = chitchat.lock().await;
+                                if expected_status.check(&chitchat_guard, &chitchat_id_clone) {
+                                    break;
+                                }
+                                if !wait_printed {
+                                    info!(node_id=%chitchat_id_clone.node_id, state_snapshot=?chitchat_guard.state_snapshot(), "Waiting for node status to be {:?}", expected_status);
+                                    wait_printed = true;
+                                }
+                                drop(chitchat_guard);
+                                tokio::time::sleep(Duration::from_millis(100)).await;
+                            }
+                        }).await.map_err(|_| {
+                            anyhow!("Node {:?} assert timeout on chitchat_id={}", expected_status, chitchat_id.node_id)
+                        }).unwrap();
+                    } else {
+                        let chitchat_guard = chitchat.lock().await;
+                        info!(node_states=?chitchat_guard.node_states(), "Node presence assert");
+                        info!(live_node_state=?chitchat_guard.live_nodes().collect_vec(), "Node presence assert");
+                        info!(dead_node_state=?chitchat_guard.dead_nodes().collect_vec(), "Node presence assert");
```

Comment:
why 3 lines? you can have more than one structure param per info log line.

Reply:
Aouch, I was meaning to remove these.